src/resilient_service.bal

```diff
@@ -1,94 +1,107 @@
 
   @http:ResourceConfig {
     methods:["GET"],
     path:"/"
   }
-  getTime (endpoint caller, http:Request req) {
+  resource function getTime(http:Caller caller, http:Request req) {
 
-    var response = legacyServiceResilientEP -> 
-        get("/legacy/localtime");
+    var response = legacyServiceResilientEP->
+                    get("/legacy/localtime");
 
-    match response {
+    if (response is http:Response) {
 
-      // Circuit breaker not tripped
-      http:Response res => {
+      // Circuit breaker not tripped.
         http:Response okResponse = new;
-        if (res.statusCode == 200) {
+        if (response.statusCode == 200) {
+
+          var payloadContent = response.getTextPayload();
+          if (payloadContent is string) {
 
-          string payloadContent = check res.getTextPayload();
-          previousRes = untaint payloadContent;
-          okResponse.setTextPayload(untaint payloadContent);
-          io:println("Remote service OK, data received");
+            previousRes = untaint payloadContent;
+            okResponse.setPayload(untaint payloadContent);
+            log:printInfo("Remote service OK, data received");
 
+          } else if (payloadContent is error) {
+
+            // Remote endpoint returns an error.
+            log:printError("Error in parsing response payload.");
+            okResponse.setPayload("Previous Response : "
+                                  + previousRes);
+          }
         } else {
 
-            // Remote endpoint returns an error
-            io:println("Error received from "+"remote service.");
-            okResponse.setTextPayload("Previous Response : "
-                + previousRes);
+            // Remote endpoint returns an error.
+            log:printError("Error received from remote service.");
+            okResponse.setPayload("Previous Response : "
+                                  + previousRes);
 
         }
         okResponse.statusCode = http:OK_200;
-        _ = caller -> respond(okResponse);
-      }
+        var result = caller->respond(okResponse);
+        handleError(result);
+
+    } else if (response is error) {
 
-      // Circuit breaker tripped and generates error
-      error err => {
+        // Circuit breaker tripped and generates error
         http:Response errResponse = new;
-        io:println("Circuit open, using cached data");
-        errResponse.setTextPayload( "Previous Response : "
-            + previousRes);
+        log:printInfo("Circuit open, using cached data");
+        errResponse.setPayload( "Previous Response : "
+                                + previousRes);
 
-        // Inform client service is unavailable
+        // Inform client service unavailability.
         errResponse.statusCode = http:OK_200;
-        _ = caller -> respond(errResponse);
-      }
+        var result = caller->respond(errResponse);
+        handleError(result);
     }
   }
 }
+
+function handleError(error? result) {
```

Comment:
don't include abstracted functions in the playground samples - it makes it harder to follow in a limited view playpen. This method is only called once so just inline it for readability.

Reply:
Simplified the code and issued a new PR >> https://github.com/ballerina-guides/playground-circuit-breaker/pull/28/files